plugin/trino-hive/src/main/java/io/trino/plugin/hive/HiveMetadata.java

```diff
@@ -2420,8 +2423,8 @@ public Optional<ConnectorViewDefinition> getView(ConnectorSession session, Schem
                         throw new HiveViewNotSupportedException(viewName);
                     }
 
-                    ConnectorViewDefinition definition = createViewReader(metastore, session, view, typeManager, this::redirectTable, metadataProvider)
-                            .decodeViewData(view.getViewOriginalText().get(), view, catalogName);
+                    ConnectorViewDefinition definition = createViewReader(metastore, session, view, typeManager, this::redirectTable, metadataProvider, true)
+                            .decodeViewData(view.getViewOriginalText().get(), view, catalogName, runHiveViewsAsInvoker);
```

Comment:
per above, pass the configuration to `createViewReader` and don't pass it to `decodeViewData()`

Reply:
Thanks, re-wrote this based on your guidance.